### PR TITLE
Fix missing sendFrame after re-adding streams

### PR DESCRIPTION
**Pull request by dnachbaur**

### CODE REVIEW

**Comment by dnachbaur** [comment]
Retest this please

**Review by rdumusc — COMMENTED** [review]
Hmmm no, I think pixelStreamOpened(uri); should really be emitted only once. I would say the problem is more with the fact that the frameDispatcher ignores the first requestFrame if there is no Stream connected.

File: deflect/FrameDispatcher.cpp [hunk 1]
```diff
@@ -172,10 +172,11 @@ void FrameDispatcher::deleteStream(const QString uri)
     if (_impl->streamBuffers[uri].getSourceCount() == 0 &&
         _impl->streamBuffers.count(uri))
     {
-        _impl->streamBuffers.erase(uri);
-
         if (_impl->observers[uri] == 0)
+        {
```
Comment (rdumusc) [inline deflect/FrameDispatcher.cpp]: move in other if

File: tests/cpp/ServerTests.cpp [hunk 2]
```diff
@@ -196,7 +196,127 @@ BOOST_AUTO_TEST_CASE(testOneObserverAndOneStream)
...
+                                   serverPort());
+            BOOST_REQUIRE(stream.isConnected());
+        }
+    }
+
+    waitForMessage();
+    BOOST_CHECK_EQUAL(getOpenedStreams(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(closeAndReopenStreamWithObserverStayingAlive)
```
Comment (rdumusc) [inline tests/cpp/ServerTests.cpp]: maybe not needed

**Review by rdumusc — APPROVED** [review]

**Pull request merged**